src/foam/u2/filter/properties/StringArrayFilterView.js

```diff
@@ -0,0 +1,186 @@
+/**
+ * @license
+ * Copyright 2025 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.u2.filter.properties',
+  name: 'StringArrayFilterView',
+  extends: 'foam.u2.filter.properties.StringFilterView',
+
+  documentation: 'Filter view for String[] properties. Builds OR(IN("x", prop), IN("y", prop), ...).',
+
+  requires: [
+    'foam.u2.CheckBox',
+    'foam.u2.TextField'
+  ],
+
+  properties: [
+    {
+      class: 'String',
+      name: 'search',
+      documentation: 'Client-side starts-with filter for option names.',
+      postSet: function() { this.daoUpdate(); }
+    },
+    { class: 'Int',     name: 'maxOptionCount', value: 20 },
+    { name: 'countByContents', factory: function() { return {}; } },
+    {
+      name: 'availableOptions',
+      expression: function(countByContents, selectedOptions) {
+        var all = Object.keys(countByContents || {});
+        if ( ! all.length ) return [];
+        var set = new Set(selectedOptions || []);
+        return all.filter(o => ! set.has(o)).sort();
+      }
+    },
+    {
+      name: 'predicate',
+      expression: function(selectedOptions, property) {
+        if ( ! selectedOptions || selectedOptions.length === 0 ) return this.TRUE;
+        if ( selectedOptions.length === 1 ) return this.IN(selectedOptions[0], property);
+        var self = this;
+        return selectedOptions.reduce(function(acc, v) {
+          var term = self.IN(v, property);
+          return acc ? self.OR(acc, term) : term;
+        }, null);
+      }
+    }
+  ],
+
+  methods: [
+    function render() {
```

Comment:
Just based on an eye scan, this whole method can be condensed to 
```
this.SUPER();
this.onDetach(this.selectedOptions$.sub(this.daoUpdate));
```
Am i wrong?

Reply:
had some minor adjustments - but no - not wrong.